### PR TITLE
Fix for defect #580

### PR DESCRIPTION
**Pull request by madhuthorat**
Signed-off-by: Madhu Thorat <madhu.punjabi@in.ibm.com>

## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?
Currently independent fileset for consistency group gets delete by CSI driver even if had not created it.

## What is the new behavior?
Independent fileset for consistency group won't get deleted by CSI driver if had not created it.

## How risky is this change?
- [x] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Review by Jainbrt — APPROVED** [review]

File: driver/csiplugin/controllerserver.go [hunk 1]
```diff
@@ -1223,27 +1258,7 @@ func (cs *ScaleControllerServer) DeleteVolume(ctx context.Context, req *csi.Dele
...
-							if err != nil {
-								glog.V(4).Infof("deletion of independent fileset for consistency group [%v] failed with error: [%v]", volumeIdMembers.ConsistencyGroup, err)
-							} else {
-								glog.V(4).Infof("deleted independent fileset for consistency group [%v]", volumeIdMembers.ConsistencyGroup)
-							}
-						}
-					} else {
-						glog.V(4).Infof("listing of filesets for filesystem: [%v] failed with error: [%v]", FilesystemName, err)
-					}
+					cs.DeleteCGFileset(FilesystemName, inodeSpace, volumeIdMembers, conn)
```
Comment (deeghuge) [inline driver/csiplugin/controllerserver.go:1261]: Shouldn't we be checking error if deletion of independent fileset failed ?

Comment (madhuthorat) [reply driver/csiplugin/controllerserver.go:1261]: Currently one reason to not handle error is that even if we fail we want to continue in this function and delete symbolic link. Another question is here we have already successfully deleted the dependent fileset for the volume and what has failed is deletion of independent fileset. If we return error here then kubernetes will keep on retrying, do we want to have that behavior ?

Comment (deeghuge) [reply driver/csiplugin/controllerserver.go:1261]: lets consider the this is last volume from CG and for some reason GUI misbehaved during deletion of independent fileset deletion and independent fileset remained there then who is going to delete it ? 

if volume deletion fails in independent fileset and csi returns error, k8s going retry again and again until all the required steps are done from deletion perspective. We should consider deletion of independent fileset as one individual step and should be retried as part of idempotency.

Comment (madhuthorat) [reply driver/csiplugin/controllerserver.go]: Done

**Review by Jainbrt — APPROVED** [review]

**Review by deeghuge — APPROVED** [review]

**Pull request merged**